extensions/ql-vscode/src/helpers.ts

```diff
@@ -47,42 +47,45 @@ export function withProgress<R>(
  * Show an error message and log it to the console
  *
  * @param message The message to show.
+ * @param logger The output logger that will receive the message
  * @param items A set of items that will be rendered as actions in the message.
  *
- * @return A thenable that resolves to the selected item or undefined when being dismissed.
+ * @return A promise that resolves to the selected item or undefined when being dismissed.
  */
-export async function showAndLogErrorMessage(message: string, ...items: string[]): Promise<string | undefined> {
-  return internalShowAndLog(message, Window.showErrorMessage, ...items);
+export async function showAndLogErrorMessage(message: string, outputLogger = logger, ...items: string[]): Promise<string | undefined> {
+  return internalShowAndLog(message, items, outputLogger, Window.showErrorMessage);
 }
 /**
  * Show a warning message and log it to the console
  *
  * @param message The message to show.
+ * @param logger The output logger that will receive the message
  * @param items A set of items that will be rendered as actions in the message.
  *
- * @return A thenable that resolves to the selected item or undefined when being dismissed.
+ * @return A promise that resolves to the selected item or undefined when being dismissed.
  */
-export async function showAndLogWarningMessage(message: string, ...items: string[]): Promise<string | undefined> {
-  return internalShowAndLog(message, Window.showWarningMessage, ...items);
+export async function showAndLogWarningMessage(message: string, outputLogger = logger, ...items: string[]): Promise<string | undefined> {
+  return internalShowAndLog(message, items, outputLogger, Window.showWarningMessage);
 }
 /**
  * Show an information message and log it to the console
  *
  * @param message The message to show.
+ * @param logger The output logger that will receive the message
  * @param items A set of items that will be rendered as actions in the message.
  *
- * @return A thenable that resolves to the selected item or undefined when being dismissed.
+ * @return A promise that resolves to the selected item or undefined when being dismissed.
  */
-export async function showAndLogInformationMessage(message: string, ...items: string[]): Promise<string | undefined> {
-  return internalShowAndLog(message, Window.showInformationMessage, ...items);
+export async function showAndLogInformationMessage(message: string, outputLogger = logger, ...items: string[]): Promise<string | undefined> {
+  return internalShowAndLog(message, items, outputLogger, Window.showInformationMessage);
 }
 
-async function internalShowAndLog(message: string, fn: Function, ...items: string[]): Promise<string | undefined> {
-  logger.log(message);
-  const label = 'Show log';
+async function internalShowAndLog(message: string, items: string[], outputLogger = logger, fn: Function): Promise<string | undefined> {
```

Comment:
Totally a nit, but I think I'd prefer the specific type `(message: string, label: string, ...items: string[]) => Thenable<string | undefined>` to just `Function`.